### ISSUE TITLE
Apply clip area on the labels/captions considering the padding

### ISSUE THREAD

**Issue by stockiNail** [kurkle/chartjs-chart-treemap#135]
Currently before drawing a caption or a label, the area of the element will be clipped.

But the full element size is clipped, instead it should clip an area considering the padding set in the configuration.

https://github.com/kurkle/chartjs-chart-treemap/blob/94efad8d9aea17668b96a28e9eae0bb4db932d7d/src/element.js#L90-L96

This is a breaking change because it could change how the labels or captions will be cut.

**Comment by kurkle** [comment]
This one could be considered a bug instead of breaking change. That said, I think it's ok to delay until v3 and then the breaking change label is fine.

**Comment by stockiNail** [comment]
Yes it could be. Sorry if I have seen only this morning. Anyway no issue till now!

**Comment by stockiNail** [comment]
@kurkle I meant that till now the current logic (without considering the padding) seems to be good for the users. This is the reason why I didn't tag as bug.

**Comment by stockiNail** [comment]
@kurkle anyway if you want I can do it as bug fixing. I have already started but I stopped me because it sounded a breaking change.
